Framework/DataObjects/src/LeanPeak.cpp

```diff
@@ -0,0 +1,314 @@
+ * No detector ID is set.
+ *
+ * @param QSampleFrame :: Q of the center of the peak, in reciprocal space
+ * @param goniometer :: a 3x3 rotation matrix
+ * @param wavelength :: wavelength in Angstroms.
+ */
+LeanPeak::LeanPeak(const Mantid::Kernel::V3D &QSampleFrame,
+                   const Mantid::Kernel::Matrix<double> &goniometer,
+                   double wavelength)
+    : BasePeak(goniometer) {
+  this->setQSampleFrame(QSampleFrame);
+  this->setWavelength(wavelength);
+}
+
+/**
+ * @brief Copy constructor
+ * @param other : Source
+ */
+LeanPeak::LeanPeak(const LeanPeak &other)
+    : BasePeak(other), m_Qsample(other.m_Qsample) {}
+
+//----------------------------------------------------------------------------------------------
+/** Constructor making a LeanPeak from IPeak interface
+ *
+ * @param ipeak :: const reference to an IPeak object
+ */
+LeanPeak::LeanPeak(const Geometry::IPeak &ipeak)
+    : BasePeak(ipeak), m_Qsample(ipeak.getQSampleFrame()) {}
+
+//----------------------------------------------------------------------------------------------
+/** Set the detector ID of the pixel at the centre of the peak and look up and
+ * cache values related to it. It also adds it to the list of contributing
+ * detectors for this peak but does NOT remove the old centre.
+ */
+void LeanPeak::setDetectorID(int) {
+  throw Exception::NotImplementedError(
+      "LeanPeak::setDetectorID(): Can't set detectorID on LeanPeak");
+}
+
+//----------------------------------------------------------------------------------------------
+/** Get the ID of the detector at the center of the peak  */
+int LeanPeak::getDetectorID() const { return -1; }
+
+//----------------------------------------------------------------------------------------------
+/** Set the instrument (and save the source/sample pos).
+ * Call setDetectorID AFTER this call.
+ *
+ */
+void LeanPeak::setInstrument(const Geometry::Instrument_const_sptr &) {
+  throw Exception::NotImplementedError(
+      "LeanPeak::setInstrument(): Can't set instrument on LeanPeak");
+}
+
+//----------------------------------------------------------------------------------------------
+/** Return a shared ptr to the detector at center of peak. */
+Geometry::IDetector_const_sptr LeanPeak::getDetector() const {
+  throw Exception::NotImplementedError(
+      "LeanPeak::getDetector(): Has no detector ID");
+}
+
+/** Return a shared ptr to the instrument for this peak. */
+Geometry::Instrument_const_sptr LeanPeak::getInstrument() const {
+  throw Exception::NotImplementedError(
+      "LeanPeak::setInstrument(): Has no instrument");
+}
+
+// -------------------------------------------------------------------------------------
+/** Calculate the time of flight (in microseconds) of the neutrons for this
+ * peak,
+ * using the geometry of the detector  */
+double LeanPeak::getTOF() const {
+  return std::numeric_limits<double>::quiet_NaN();
```

Comment:
Why does this return nan when other invalid operations throw exceptions?

Reply:
So this method and `getDetectorID` are needed to return values because they are a column of the peaks table workspace. I have a separate issue, SCD197, to customise the LeanPeaksWorkspace (once implemented) to only include the columns we have. I added changing these methods to the task.

Reply:
https://github.com/mantidproject/mantid/blob/7645f6210d846f934945c61d1c90e8f0dd5dd943/Framework/DataObjects/src/Peak.cpp#L979